dialectic-reconnect/src/retry.rs

```diff
@@ -0,0 +1,522 @@
+//! The *retry* end of a reconnectable backend consists of a [`Connector`] and the set of [`Chan`]s
+//! spawned from it with [`connect`](Connector::connect), which automatically reconnect to their
+//! other side when they encounter an error.
+
+use dialectic::{
+    backend::{self, By},
+    prelude::*,
+};
+use std::{
+    fmt::{Debug, Display},
+    future::Future,
+    marker::PhantomData,
+    pin::Pin,
+    sync::Arc,
+    time::Duration,
+};
+use tokio::time::Instant;
+
+/// A description of what to do when an error happens that we might want to recover from.
+#[derive(Debug, Clone, Copy)]
+pub enum RetryStrategy {
+    /// Pause for this amount of time, then retry without discarding the current connection.
+    RetryAfter(Duration),
+    /// Discard the current connection, then retry after this amount of time.
+    ReconnectAfter(Duration),
```

Comment:
Almost always, you want `ReconnectAfter`. The only reason you want `RetryAfter` is if the error is something you expect to be transient, which won't be thrown again if you retry the same operation on the same underlying connection. I'm not sure what backend errors would have this property, but it was easy enough to implement, so I put it in. If we removed this option, the interface could be simpler. Maybe we should take it out?

Reply:
> If we removed this option, the interface could be simpler. Maybe we should take it out?

Ahh, yes, that makes sense. If we removed it, would it be impossible to do the `RetryAfter` behavior or would it just be inconvenient?

Reply:
It would be impossible, because we would always destroy the connection end, rather than keeping it. But, I'm really not sure what kinds of connection errors ought to be handled by a retry rather than a reconnect. Can you think of any?